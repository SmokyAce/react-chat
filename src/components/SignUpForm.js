import React from 'react';
import { withStyles } from 'material-ui/styles';
import TextField from 'material-ui/TextField';
import Button from 'material-ui/Button';

const styles = theme => ({
  container: {
    display: 'flex',
    flexWrap: 'wrap',
    flexDirection: 'column',
  },
  button: {
    marginTop: theme.spacing.unit * 2,
  },
});

class SignUpForm extends React.Component {
  state = {
    username: {
      value: '',
      isValid: true,
    },
    password: {
      value: '',
      isValid: true,
    },
    repeatedPassword: {
      value: '',
      isValid: true,
    },
  };

  handleInputChange = (event) => {
    event.persist();
    const { name, value } = event.target;
    this.setState(prevState => ({
      [name]: {
        ...prevState[name],
        value,
      },
    }));
  };

  validate = () => {
    const { password: { value: pwd }, repeatedPassword: { value: repeatPwd } } = this.state;

    const isValid = pwd === repeatPwd;

    this.setState(prevState => ({
      password: {
        ...prevState.password,
        isValid,
      },
      repeatedPassword: {
        ...prevState.repeatedPassword,
        isValid,
      },
    }));

    return isValid;
  };

  handleSubmit = (event) => {
    event.preventDefault();

    if (!this.validate()) {
      return;
    }

    const { username, password } = this.state;

    this.props.onSubmit(username.value, password.value);
  };

  render() {
    const { classes } = this.props;
    const { username, password, repeatedPassword } = this.state;

    return (
      <form className={classes.container} onSubmit={this.handleSubmit}>
        <TextField
          fullWidth
          required
          label="Username"
          placeholder="Type your username ..."
          name="username"
          margin="normal"
          username={username.value}
          onChange={this.handleInputChange}
          error={!username.isValid}
        />
        <TextField
          fullWidth
          required
          label="Password"
          placeholder="Type your password ..."
          name="password"
          type="password"
          margin="normal"
          username={password.value}
          onChange={this.handleInputChange}
          error={!password.isValid}
        />
        <TextField
          fullWidth
          required
          label="Repeat password"
          placeholder="Repeat your password ..."
          name="repeatedPassword"
          type="password"
          margin="normal"
          username={repeatedPassword.value}
          onChange={this.handleInputChange}
          error={!repeatedPassword.isValid}
        />
        <Button variant="raised" color="primary" className={classes.button} type="submit">
          SignUp
        </Button>
      </form>
    );
  }
}

export default withStyles(styles)(SignUpForm);
