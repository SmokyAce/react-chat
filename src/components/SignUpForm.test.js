/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import SignUpForm from './SignUpForm';

const mockProps = {
  onSubmit: jest.fn(),
};

describe('<SignUpForm />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SignUpForm {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<SignUpForm {...mockProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
