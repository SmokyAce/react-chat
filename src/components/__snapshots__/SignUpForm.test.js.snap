// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpForm /> renders correctly 1`] = `
<form
  className="SignUpForm-container-57"
  onSubmit={[Function]}
>
  <div
    aria-describedby={undefined}
    className="MuiFormControl-root-59 MuiFormControl-marginNormal-60 MuiFormControl-fullWidth-62"
    onBlur={[Function]}
    onFocus={[Function]}
    username=""
  >
    <label
      className="MuiFormLabel-root-68 MuiInputLabel-root-63 MuiInputLabel-formControl-64 MuiInputLabel-animated-67"
      data-shrink={false}
      htmlFor={undefined}
    >
      Username
      <span
        className="MuiFormLabel-asterisk-72"
      >
         *
      </span>
    </label>
    <div
      className="MuiInput-root-74 MuiInput-fullWidth-81 MuiInput-formControl-75 MuiInput-underline-78"
    >
      <input
        aria-invalid={false}
        aria-required={true}
        autoComplete={undefined}
        autoFocus={undefined}
        className="MuiInput-input-82"
        defaultValue={undefined}
        disabled={false}
        id={undefined}
        name="username"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder="Type your username ..."
        readOnly={undefined}
        required={true}
        rows={undefined}
        type="text"
        value={undefined}
      />
    </div>
  </div>
  <div
    aria-describedby={undefined}
    className="MuiFormControl-root-59 MuiFormControl-marginNormal-60 MuiFormControl-fullWidth-62"
    onBlur={[Function]}
    onFocus={[Function]}
    username=""
  >
    <label
      className="MuiFormLabel-root-68 MuiInputLabel-root-63 MuiInputLabel-formControl-64 MuiInputLabel-animated-67"
      data-shrink={false}
      htmlFor={undefined}
    >
      Password
      <span
        className="MuiFormLabel-asterisk-72"
      >
         *
      </span>
    </label>
    <div
      className="MuiInput-root-74 MuiInput-fullWidth-81 MuiInput-formControl-75 MuiInput-underline-78"
    >
      <input
        aria-invalid={false}
        aria-required={true}
        autoComplete={undefined}
        autoFocus={undefined}
        className="MuiInput-input-82 MuiInput-inputType-86"
        defaultValue={undefined}
        disabled={false}
        id={undefined}
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder="Type your password ..."
        readOnly={undefined}
        required={true}
        rows={undefined}
        type="password"
        value={undefined}
      />
    </div>
  </div>
  <div
    aria-describedby={undefined}
    className="MuiFormControl-root-59 MuiFormControl-marginNormal-60 MuiFormControl-fullWidth-62"
    onBlur={[Function]}
    onFocus={[Function]}
    username=""
  >
    <label
      className="MuiFormLabel-root-68 MuiInputLabel-root-63 MuiInputLabel-formControl-64 MuiInputLabel-animated-67"
      data-shrink={false}
      htmlFor={undefined}
    >
      Repeat password
      <span
        className="MuiFormLabel-asterisk-72"
      >
         *
      </span>
    </label>
    <div
      className="MuiInput-root-74 MuiInput-fullWidth-81 MuiInput-formControl-75 MuiInput-underline-78"
    >
      <input
        aria-invalid={false}
        aria-required={true}
        autoComplete={undefined}
        autoFocus={undefined}
        className="MuiInput-input-82 MuiInput-inputType-86"
        defaultValue={undefined}
        disabled={false}
        id={undefined}
        name="repeatedPassword"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder="Repeat your password ..."
        readOnly={undefined}
        required={true}
        rows={undefined}
        type="password"
        value={undefined}
      />
    </div>
  </div>
  <button
    className="MuiButtonBase-root-103 MuiButton-root-88 MuiButton-raised-93 MuiButton-raisedPrimary-95 SignUpForm-button-58"
    disabled={false}
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="submit"
  >
    <span
      className="MuiButton-label-89"
    >
      SignUp
    </span>
    <span
      className="MuiTouchRipple-root-106"
    />
  </button>
</form>
`;
